samples/tutorials/hardwareInterfaces/i2c/BatteryChargerReading/batteryReadingComponent/batteryReading.c

```diff
@@ -0,0 +1,275 @@
+        return LE_FAULT;
+    }
+
+    return fd;
+}
+
+//--------------------------------------------------------------------------------------------------
+/**
+ * Performs an SMBUS read of a 1 byte register
+ *
+ * @return
+ *      - LE_OK
+ *      - LE_FAULT
+ */
+//--------------------------------------------------------------------------------------------------
+static le_result_t SmbusReadReg
+(
+    uint8_t i2cBus,  ///< [IN] I2C bus to perform the read on
+    uint8_t i2cAddr, ///< [IN] I2C address to read from
+    int8_t  reg,     ///< [IN] Register within the I2C device to read
+    uint8_t *data    ///< [OUT] Value stored within the register.  This value is only valid if the
+                     ///  function returned LE_OK.
+)
+{
+    int i2cFd = I2cAccessBusAddr(i2cBus, i2cAddr);
+    if (i2cFd == LE_FAULT)
+    {
+        LE_ERROR("failed to open i2c bus %d for access to address %d\n", i2cBus, i2cAddr);
+        return  LE_FAULT;
+    }
+
+    le_result_t result;
+
+    const int readResult = i2c_smbus_read_byte_data(i2cFd, reg);
+
+    if (readResult < 0)
+    {
+        LE_ERROR("smbus read failed with error %d", readResult);
+        result = LE_FAULT;
+    }
+    else
+    {
+        *data = readResult;
+        result = LE_OK;
+    }
+
+    close(i2cFd);
+
+    LE_DEBUG("SMBUS READ addr=0x%x, reg=0x%x, data=0x%x", i2cAddr, reg, *data);
+
+    return result;
+}
+
+//--------------------------------------------------------------------------------------------------
+/**
+ * Performs an SMBUS write of a 1 byte register
+ *
+ * @return
+ *      - LE_OK
+ *      - LE_FAULT
+ */
+//--------------------------------------------------------------------------------------------------
+static le_result_t SmbusWriteReg
+(
+    uint8_t i2cBus,   ////< I2C bus to perform the write on
+    uint8_t i2cAddr,  ////< I2C device  address 
+    uint8_t reg,      ////< Register within the I2C device to write to
+    uint8_t data      ////< Data to write to the given register
+)
+{
+int i2cFd = I2cAccessBusAddr(i2cBus, i2cAddr);
+
+if (i2cFd == LE_FAULT)
+{
+    LE_ERROR("failed to open i2c bus %d for access to address %d\n", i2cBus, i2cAddr);
+    return LE_FAULT ;
+}
+
+le_result_t result;
+
+const int writeResult = i2c_smbus_write_byte_data(i2cFd, reg, data);
+if (writeResult < 0)
+{
+    LE_ERROR("smbus write failed with error %d", writeResult);
+    result = LE_FAULT;
+}
+else
+{
+    result = LE_OK;
+}
+
+close(i2cFd);
+
+LE_DEBUG("SMBUS Write addr 0x%x, reg=0x%x, data=0x%x", i2cAddr, reg, data);
+
+return result;
+}
+
+/**-----------------------------------------------------------------------------------------------
+ * Enables I2C Switch
+ *
+ * 
+ *      
+ *      
+ */
+//--------------------------------------------------------------------------------------------------
+static void EnableI2cBus
+(
+    void
+)
+{
+    LE_DEBUG("Enabling TCA9546A I2C switch...");
```

Comment:
Maybe we should have some comments indicating why we need to enable the i2c switch.